common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResultFactory.java

```diff
@@ -0,0 +1,132 @@
+                    return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                            exception.getErrorCode(), exception.getMessage());
+                }
+
+            case AuthenticationConstants.UIResponse.BROKER_REQUEST_RESUME:
+                Logger.verbose(TAG, "Device needs to have broker installed, we expect the apps to call us"
+                        + "back when the broker is installed");
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                        MicrosoftAuthorizationErrorResponse.AUTHORIZATION_FAILED,
+                        MicrosoftAuthorizationErrorResponse.BROKER_NEEDS_TO_BE_INSTALLED);
+        }
+        return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                MicrosoftAuthorizationErrorResponse.UNKNOWN_ERROR,
+                MicrosoftAuthorizationErrorResponse.UNKNOWN_RESULT_CODE + "[" + resultCode + "]");
+    }
+
+    private AzureActiveDirectoryAuthorizationResult createAuthorizationResultWithErrorResponse(final AuthorizationStatus authStatus,
+                                                                                               final String error,
+                                                                                               final String errorDescription) {
+        AuthorizationErrorResponse errorResponse = new AzureActiveDirectoryAuthorizationErrorResponse(error, errorDescription);
+        return new AzureActiveDirectoryAuthorizationResult(authStatus, errorResponse);
+    }
+
+    private AzureActiveDirectoryAuthorizationResult createAuthorizationResultWithErrorResponse(final AuthorizationStatus authStatus,
+                                                                                               final String error,
+                                                                                               final String errorDescription,
+                                                                                               final String errorCodes,
+                                                                                               final String correlationId) {
+        Logger.info(TAG, correlationId, "Error is returned from webview redirect");
+        Logger.infoPII(TAG, correlationId, "error: " + error + " errorDescription: " + errorDescription);
+        AzureActiveDirectoryAuthorizationErrorResponse errorResponse =
+                new AzureActiveDirectoryAuthorizationErrorResponse(error, errorDescription);
+        errorResponse.setErrorCodes(errorCodes);
+        return new AzureActiveDirectoryAuthorizationResult(authStatus, errorResponse);
+    }
+
+    private AzureActiveDirectoryAuthorizationResult parseUrlAndCreateAuthorizationResult(final String url) {
+        HashMap<String, String> urlParameters = StringExtensions.getUrlParameters(url);
```

Comment:
Looks like this could be declared `final`

Reply:
Done.